In [1]:
# Environment Setting

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/titanic/train_and_test2.csv


*Importing Required Libraries*

In [2]:
import pandas as pd

In [3]:
df = pd.read_csv("/kaggle/input/titanic/train_and_test2.csv")

In [4]:
df.head()

Passengerid   Age     Fare  Sex  sibsp  zero  zero.1  zero.2  zero.3  \
0            1  22.0   7.2500    0      1     0       0       0       0   
1            2  38.0  71.2833    1      1     0       0       0       0   
2            3  26.0   7.9250    1      0     0       0       0       0   
3            4  35.0  53.1000    1      1     0       0       0       0   
4            5  35.0   8.0500    0      0     0       0       0       0   

   zero.4  ...  zero.12  zero.13  zero.14  Pclass  zero.15  zero.16  Embarked  \
0       0  ...        0        0        0       3        0        0       2.0   
1       0  ...        0        0        0       1        0        0       0.0   
2       0  ...        0        0        0       3        0        0       2.0   
3       0  ...        0        0        0       1        0        0       2.0   
4       0  ...        0        0        0       3        0        0       2.0   

   zero.17  zero.18  2urvived  
0        0        0         0  
1        0        0         1  
2        0        0         1  
3        0        0         1  
4        0        0         0  

[5 rows x 28 columns]

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 28 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Passengerid  1309 non-null   int64  
 1   Age          1309 non-null   float64
 2   Fare         1309 non-null   float64
 3   Sex          1309 non-null   int64  
 4   sibsp        1309 non-null   int64  
 5   zero         1309 non-null   int64  
 6   zero.1       1309 non-null   int64  
 7   zero.2       1309 non-null   int64  
 8   zero.3       1309 non-null   int64  
 9   zero.4       1309 non-null   int64  
 10  zero.5       1309 non-null   int64  
 11  zero.6       1309 non-null   int64  
 12  Parch        1309 non-null   int64  
 13  zero.7       1309 non-null   int64  
 14  zero.8       1309 non-null   int64  
 15  zero.9       1309 non-null   int64  
 16  zero.10      1309 non-null   int64  
 17  zero.11      1309 non-null   int64  
 18  zero.12      1309 non-null   int64  
 19  zero.1

In [6]:
df = df[["Pclass", "Age", "Fare", "Sex", "2urvived"]]

In [7]:
df.columns = ["Pclass", "Age", "Fare", "Sex", "Survived"]

In [8]:
df.head()

Pclass   Age     Fare  Sex  Survived
0       3  22.0   7.2500    0         0
1       1  38.0  71.2833    1         1
2       3  26.0   7.9250    1         1
3       1  35.0  53.1000    1         1
4       3  35.0   8.0500    0         0

In [9]:
df.isnull().sum()

Pclass      0
Age         0
Fare        0
Sex         0
Survived    0
dtype: int64

In [10]:
X = df.iloc[:,:-1]
y = df.iloc[:,4]

In [11]:
X.head()

Pclass   Age     Fare  Sex
0       3  22.0   7.2500    0
1       1  38.0  71.2833    1
2       3  26.0   7.9250    1
3       1  35.0  53.1000    1
4       3  35.0   8.0500    0

In [12]:
y.head()

0    0
1    1
2    1
3    1
4    0
Name: Survived, dtype: int64

*Splitting the dataset into train and test data*

In [13]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.33, random_state=42)

In [14]:
from sklearn.preprocessing import StandardScaler

In [15]:
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)

In [16]:
X_train_scaled

array([[ 0.85920374, -0.60169694, -0.52828855,  1.37636821],
       [ 0.85920374, -0.12437351, -0.38428307,  1.37636821],
       [ 0.85920374, -0.68125085, -0.51230637, -0.72654977],
       ...,
       [-1.5103613 , -0.99946647,  0.26782405, -0.72654977],
       [ 0.85920374,  0.90982727, -0.39127528, -0.72654977],
       [ 0.85920374, -0.76080475, -0.51621801, -0.72654977]])

In [17]:
X_test.head()

Pclass   Age    Fare  Sex
1148       3  28.0   8.050    0
1049       1  42.0  26.550    0
982        3  28.0   7.775    0
808        2  39.0  13.000    0
1195       3  28.0   7.750    1

In [18]:
X_test_scaled = scaler.transform(X_test)

In [19]:
X_test_scaled

array([[ 0.85920374, -0.12437351, -0.51230637, -0.72654977],
       [-1.5103613 ,  0.98938117, -0.14271833, -0.72654977],
       [ 0.85920374, -0.12437351, -0.51780024, -0.72654977],
       ...,
       [ 0.85920374,  0.07451126, -0.51230637, -0.72654977],
       [ 0.85920374, -0.91991256, -0.51230637,  1.37636821],
       [ 0.85920374,  0.75071946, -0.09127567,  1.37636821]])

In [20]:
from sklearn.linear_model import LogisticRegression
classification = LogisticRegression()

In [21]:
classification.fit(X_train_scaled, y_train)

LogisticRegression()

In [22]:
ypred = classification.predict(X_test_scaled)

In [23]:
ypred

array([0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0,
       1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0,
       1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0,
       0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1,
       0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1,
       0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0,
       0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1,

In [24]:
from sklearn.metrics import classification_report, confusion_matrix

In [25]:
print(classification_report(y_test, ypred))

              precision    recall  f1-score   support

           0       0.81      0.93      0.87       318
           1       0.67      0.39      0.49       114

    accuracy                           0.79       432
   macro avg       0.74      0.66      0.68       432
weighted avg       0.77      0.79      0.77       432



In [26]:
confusion_matrix(y_test, ypred)

array([[296,  22],
       [ 70,  44]])

*accuracy=(296+44)/(296+22+70+44) 
=>0.78703
=>~0.79.*
*hence, we got accuracy of 0.79 for the Logistic Regression model.*